precompiles/xcm/XCM.sol

```diff
@@ -21,48 +26,22 @@ interface XCM {
     function assets_withdraw(
         address[] calldata asset_id,
         uint256[] calldata asset_amount,
-        bytes32   recipient_account_id,
-        bool      is_relay,
-        uint256   parachain_id,
-        uint256   fee_index
-    ) external returns (bool);
-
-    /**
-     * @dev Withdraw assets using PalletXCM call.
-     * @param asset_id - list of XC20 asset addresses
-     * @param asset_amount - list of transfer amounts (must match with asset addresses above)
-     * @param recipient_account_id - ETH address of the destination account
-     * @param is_relay - set `true` for using relay chain as reserve
-     * @param parachain_id - set parachain id of reserve parachain (when is_relay set to false)
-     * @param fee_index - index of asset_id item that should be used as a XCM fee
-     * @return bool confirmation whether the XCM message sent.
-     *
-     * How method check that assets list is valid:
-     * - all assets resolved to multi-location (on runtime level)
-     * - all assets has corresponded amount (lenght of assets list matched to amount list)
-     */
-    function assets_withdraw(
-        address[] calldata asset_id,
-        uint256[] calldata asset_amount,
-        address   recipient_account_id,
-        bool      is_relay,
-        uint256   parachain_id,
+        Multilocation memory beneficiary,
+        Multilocation memory destination,
```

Comment:
What happens with all the existing smart contracts which rely on the old interface?

Reply:
so should I add all the previous function of backward compatibility and rename these new functions?

Reply:
Yes, you need to keep the old interface as is. We cannot remove support for it.

And I don't believe you need to rename any functions, since Solidity supports function overload.

I'd suggest to keep the old `XCM.sol` file as is, and create a new one called `XCM_v2.sol`, adding a disclaimer we encourage smart contracts to use the 2nd one.